# topics
speech_service: "/suitbot/audio/speech_output"
listening_service: "/suitbot/audio/set_listening"
usr_cmd_topic: "/suitbot/audio/cmd_in"

ctrl_topic: "/suitbot/ctrl/velocity"
force_topic: "/suitbot/handle/force"
encoder_topic: "/suitbot/mobility/velocity"
voltage_topic: "/suitbot/voltage"
drive_status_topic: "/suitbot/mobility/status"

reset_path_service: "/suitbot/ctrl/reset_course"
planner_arrow_topic: "/suitbot/planner/arrow"
global_map_topic: "/suitbot/planner/global_map"
planned_path_topic: "/suitbot/planner/planned_path"

lidar_original_topic: "/livox/lidar"
lidar_sync_topic: "/suitbot/lidar"

pf_particles_topic: "/suitbot/localization/particles"
pcd_map_topic: "/suitbot/localization/pcd_map"

local_map_topic: "/suitbot/planner/local_map"
local_map_image_topic: "/suitbot/planner/local_map_image"

# for testing, decoupling audio
course_idx: 2

# for user cmd
states_map:
  - key: 0 
    name: "cancel"
    pos: [0,0,0]
  - key: 2
    name: "left"
    pos: [0,0,0]
  - key: 3
    name: "middle" 
    pos: [0,0,0]
  - key: 4
    name: "right" 
    pos: [0,0,0]
  - key: 5
    name: "nothing"
    pos: [0,0,0]
  - key: 6
    name: "elevator" 
    pos: [207,183,0]
  - key: 7
    name: "nineteen" 
    pos: [137,360,0]
  - key: 8
    name: "four" 
    pos: [215,395,0]
  - key: 9
    name: "six" 
    pos: [211,252,0]
  - key: 10
    name: "fountain" 
    pos: [162,171,0]
  - key: 11
    name: "staircase" 
    pos: [200,436,0]

keywords_map:
  - left
  - middle
  - right
  - nothing
  - cancel
  - terminate 
  - end
  - stop
  - elevator
  - nineteen
  - four
  - six
  - fountain
  - staircase

global_frame_id: "world"

voxel_size: 0.1

fixed_height: 0.4  # this is the height of the lidar wrt the ground

pf_update_rate: 10.0
pf_resample_interval: 0.5
pf_update_dist_threshold: 1.0
pf_update_angle_threshold: 0.3


# pf motion model parameters
pf_alpha1: 0.001
pf_alpha2: 0.0001
pf_alpha3: 0.008
pf_alpha4: 0.008
# pf sensor model parameters
pf_sigma_hit: 50.0
pf_lambda_short: 0.1
pf_max_range: 150.0
pf_max_span: 5.0
pf_z_hit: 10.0
pf_z_short: 0.1
pf_z_max: 0.1
pf_z_rand: 100.0

# pcd map transformation
tx: 0.0 
ty: 0.0
tz: 0.0
qx: 0.0
qy: 0.0
qz: 0.0
qw: 1.0


# local map parameters
local_map_xmin: -10.0
local_map_xmax: 10.0
local_map_ymin: -10.0
local_map_ymax: 10.0
local_map_zmin: -2.0
local_map_zmax: 10.0

local_icp_dt_thresh: 1.0
local_icp_dq_thresh: 0.5

local_map_lookahead: 8.0

global_map_resolution: 0.25
local_map_resolution: 0.25

obstacle_zmin: 0.3
obstacle_zmax: 1.5

pf_init_num_particles_per_grid: 3
pf_init_num_particles_total: 20000

publish_point_cloud_rate: 1.0
